### PR TITLE
Feature/swagger content

### PR DESCRIPTION
**Pull request by redhug1**
### What

Add content structs and swagger definitions

### How to review

Check swagger endpoint: /topics/{id}/content
matches to the go struct: PublicContent  and the struct's it contains

### Who can review

Anyone but me.

### CODE REVIEW

File: models/content.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	TotalCount   int                   `json:"total_count"`
+	PrivateItems []*PrivateContentItem `json:"items"`
+}
+
+// ContentItem is an individual content item
+type ContentItem struct {
+	Title string        `json:"title,omitempty"`
+	Type  string        `json:"type,omitempty"`
+	Links *ContentLinks `json:"links,omitempty"`
+	State *bool         `json:"state,omitempty"` //!!! Eleanor ... do we need something like this ?
```
Comment (DavidSubiros) [inline models/content.go:67]: is this something that @eldeal  should check? Or can we approve it for now and change it later if required?

File: swagger.yaml [hunk 2]
```diff
@@ -138,6 +189,16 @@ definitions:
...
+    description: "A link to the topic for this content"
+    properties:
+      href:
+        type: string
+        description: "The URL to the topic"
+      id:
+        type: string
+        description: "The id for the topic that this content is associated with"
   Count:
     type: string
```
Comment (DavidSubiros) [inline swagger.yaml:203]: I noticed that the limit, offset and count are defined as strings (`type : string`), I know it was already like this before this PR, but could you replace them for `type: integer` please?

**Review by DavidSubiros — COMMENTED** [review]
a very minor comment and question.

File: models/content.go (hunk 1 above)
Comment (redhug1) [reply models/content.go:67]: This can be aprovved for now and its 'up for adjustment' later when Eleanor is back.

File: swagger.yaml (hunk 2 above)
Comment (redhug1) [reply swagger.yaml:203]: Yes, now fixed

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**